app/dbConnection/index.js

```diff
@@ -220,6 +235,72 @@ exports.writeRelationships = function (relationJSON) {
     });
   });
 };
+
+/**
+ * Try to link a relationship entity with the corresponding real entity.
+ *
+ * Order:
+ * - Arists
+ * - Works
+ * - Instruments
+ *
+ * @param relEntity relationship entity object from sequelize
+ */
+function linkEntity (relEntity) {
+  // try to find something like this in the other tables
+  let artists = context.models.artists;
+  let works = context.models.works;
+  let instruments = context.models.instruments;
+  Promise.all([
+    // try artists
+    artists.findAll({
+      where: {
+        name: relEntity.name
+      }
+    }),
+    // try works
+    works.findAll({
+     where: {
+       title: relEntity.name
+     }
+    }),
+    // try instruments
+    instruments.findAll({
+      where: {
+        name: relEntity.name
+      }
+    })
+  ]).then(data => {
+    let foundArtists = data[0];
+    let foundWorks = data[1];
+    let foundInstruments = data[2];
+    if (foundArtists.length === 1) {
+      // one artist found, link it
+      foundArtists[0].getEntity().then(dbEntity => {
+        relEntity.setEntity(dbEntity);
+      });
+    } else if (foundArtists.length > 1) {
+      // more artists found, better not link
+    } else if (foundWorks.length === 1) {
+      // no artist found, but a work
+      foundWorks[0].getEntity().then(dbEntity => {
+        relEntity.setEntity(dbEntity);
+      });
+    } else if (foundWorks.length > 1) {
+      // more works found, better not link
+    } else if (foundInstruments.length === 1) {
+      // no artists and no works found, but one instrument
+      foundInstruments[0].getEntity().then(dbEntity => {
+        relEntity.setEntity(dbEntity);
+      });
+    } else if (foundInstruments.length > 1) {
+      // more instruments found, better not link
+    } else {
+      // we found nothing, no linking at all
+    }
+  });
+}
+
 exports.writeEvents = function (eventEntityJSON) {
```

Comment:
@ansjin I do not understand this method, can you check that again please?
Problems:
- The stuff i wrote in other comments
- In the last then, we collect all promises, but then ignore them?
- We should create a event, even if we have no corresponding artist or entity.
- We should not create artists!
- We should not create entities? Or should we? Makes no sense in my Opinion.

I think the best way would be to create `relationshipEntities`. Maybe.

We should check if `entityName` is `undefined` or `null`, before creating null-entries ;)
Where do we get the `entityName`? Does the algorithm already know the entity?